src/mlpack/core/dists/discrete_distribution.hpp

```diff
@@ -153,12 +202,18 @@ class DiscreteDistribution
    *    actually from this distribution.
    */
   void Train(const arma::mat& observations,
-             const arma::vec& probabilities);
+             const arma::mat& probabilities);
 
   //! Return the vector of probabilities.
-  const arma::vec& Probabilities() const { return probabilities; }
+  const arma::vec& Probabilities() const { return probabilities[0]; }
   //! Modify the vector of probabilities.
-  arma::vec& Probabilities() { return probabilities; }
+  arma::vec& Probabilities() { return probabilities[0]; }
+
+  //! Modify the vector Martix of probabilities.
+  const std::vector<arma::vec>& MultipleProbabilities() const { return probabilities; }
+
+  //! Return the vector Martix of probabilities.
+  std::vector<arma::vec>& MultipleProbabilities() { return probabilities; }
```

Comment:
>const arma::vec& Probabilities(const size_t dim = 0)
>{
> return probabilities[I];
>}

Is that mean we could keep the `MultipleProbabilities()`? If not, what should we do if we want to init the discrete distribution in the first place like https://github.com/mlpack/mlpack/pull/830/files#diff-ad93795d91105ba0fcc6e9815d5e9ad0R163?